src/dependency_specifications.rb

```diff
@@ -0,0 +1,31 @@
+
+# frozen_string_literal: true
+
+module Metalware
+  # Class to generate reused `dependency_hash`s, for use by `Dependency` object
+  # to enforce command dependencies.
+  # XXX Consider moving generating of more `dependency_hash`s here.
+  class DependencySpecifications
+    def initialize(config)
+      @config = config
+    end
+
+    def for_node_in_configured_group(node_name)
+      primary_group = configured_primary_group_for_node(node_name)
+      {
+        repo: ['configure.yaml'],
+        configure: ['domain.yaml', "groups/#{primary_group}.yaml"],
```

Comment:
I tried to make this change but I get the following error when I do so and then run a command using the dependencies:
```
[root@controller(dev) metalware]# bin/metal view-answers node node02
error: Dependency values must be an array, check: optional. Use --trace to view backtrace
```

Since this is the only thing blocking this I'm going to merge it as is, but when you've looked at that error we can add in that change; it's in https://github.com/alces-software/metalware/tree/wip-adding-optional-dependencies. Cards for this: https://trello.com/c/fnNhvA22/153-fix-specifying-optional-dependencies (for you to look at) and https://trello.com/c/2tL6Q2CP/154-add-in-optional-dependency-for-nodes-on-node-answers (to add my change in once that's done).